[Blog post] What is `this` inside foo.bar() ?

bendtherules opened the pull request:
Title - What is `this` inside foo.bar()? 

Added post content, cover image and metadata in the folder `js/this-inside-foo-bar`.

bendtherules committed:
[this-inside-foo-bar] - Add blog post and metadata

bendtherules committed:
[this-inside-foo-bar] - Fix blog title

itaditya commented on js/this-inside-foo-bar/post.md:
```diff
@@ -0,0 +1,81 @@
+
+## The problem
+
+![Code showing - method which is passed as callback loses reference to original obj as this, but calling obj.method() directly works](https://cdn.hashnode.com/res/hashnode/image/upload/v1584584613992/rwNrFCElj.png)
```
In the image it should be `onClick` instead of `.onClick` and it would be better if its called in next line

bendtherules replied:
Makes sense. Changed image and code to make evt handler usage and implementation clearer.

itaditya commented on js/this-inside-foo-bar/post.md:
```diff
@@ -0,0 +1,81 @@
+*What* →  foo is passed as `this` inside bar  
+*When* →  if bar is a function and *accessed as object property* (i.e. foo.bar).  
+
+##### For functions, 
```
adding example would be better. You can also add that when function is called in global context, `this` refers to `window`

bendtherules replied:
this value mentioned here was wrong. Fixed that.

itaditya commented on js/this-inside-foo-bar/post.md:
```diff
@@ -0,0 +1,81 @@
+
+----
+
+*That's all! Hope you had a interesting read.  
```
italics are not preferred these days.

bendtherules replied:
Changed now.

itaditya reviewed (changes requested):
some minor comments. This blog would be an interesting read. Kudos!

bendtherules committed:
[this-inside-foo-bar] post - Update problem image to show onClick details

bendtherules committed:
[this-inside-foo-bar] post - Sentence changes

bendtherules committed:
[this-inside-foo-bar] post - Change bound function link

bendtherules committed:
[this-inside-foo-bar] Change h4 subheadings to h2

bendtherules committed:
[this-inside-foo-bar] Fix this value inside function calls

bendtherules committed:
[this-inside-foo-bar] - Modify method call desc

bendtherules committed:
[this-inside-foo-bar] Modify method to fn explaination

bendtherules committed:
[this-inside-foo-bar] Modify spec desc

Add section for PropertyRef and Steps, add img for ProRef from slide, [[call]] vs .call diff

bendtherules committed:
[this-inside-foo-bar] Change footer text

bendtherules committed:
[this-inside-foo-bar] - Remove note on evt handler explicit this

bendtherules committed:
[this-inside-foo-bar] Add youtube video for spec discussion

bendtherules committed:
[this-inside-foo-bar] Add fullstop

bendtherules committed:
[this-inside-foo-bar] Refactor section on arrow and bound fn

bendtherules committed:
[this-inside-foo-bar] Adjust text hierarchy in spec section

bendtherules committed:
[this-inside-foo-bar] Reword problem explaination

bendtherules committed:
[this-inside-foo-bar] Fix language typo

bendtherules committed:
[this-inside-foo-bar] Fix in other words this value

bendtherules committed:
[this-inside-foo-bar] Change going to go

bendtherules committed:
[this-inside-foo-bar] Add section break after video

bendtherules commented:
Changes done from my side now.
Fixed a few sections and added a little more content.

bendtherules commented:
Submitted for publication - https://hashnode.com/edit/ck8d4todk00t4xes1fr1o0ena.
Please approve.
